### PR TITLE
Upgrade scalariform to 0.1.8

### PR DESCRIPTION
**Pull request by wsargent**
Upgrades Scalariform to 0.1.8.  See https://github.com/scala-ide/scalariform#usage-within-a-project

This should fix the parser errors we were getting earlier:

```
[warn] Scalariform parser error for /home/travis/build/playframework/playframework/framework/src/play-specs2/src/main/scala/play/api/test/Specs.scala: Expected token RBRACKET but got Token(XML_START_OPEN,<,3066,<)
```

### CODE REVIEW

**Comment by marcospereira** [comment]
See #6111.

Scalariform 0.1.8 has different rules, so build to [check code standards](https://travis-ci.org/playframework/playframework/jobs/151865440) fails.

**Comment by wsargent** [comment]
Okay, this is awkward -- part of the problem is that upgrading scalariformFormat to 0.1.8 makes it actually work, which means a bunch of classes that weren't previously formatted due to parsing errors now parse and format correctly.

Which means 66 files get reformatted at once.

**Comment by wsargent** [comment]
Nope, it's scalariform settings having different defaults.  See https://github.com/scala-ide/scalariform/issues/142

**Comment by schmitch** [comment]
At least one thing looks working now and that is spaces (or not) between multiple imports at the moment it didn't got changed by scalariform so you can just do whatever you like.

**Comment by wsargent** [comment]
We're down to some whitespace changes and that's it.  Think this is ready for merge.

File: framework/project/plugins.sbt [hunk 1]
```diff
@@ -19,16 +19,16 @@ scalacOptions ++= Seq("-deprecation", "-language:_")
...
 addSbtPlugin("pl.project13.scala" % "sbt-jmh" % "0.2.7")
 
 libraryDependencies ++= Seq(
   "org.scala-sbt" % "scripted-plugin" % sbtVersion.value,
   "org.webjars" % "webjars-locator-core" % "0.26"
 )
 
-// override scalariform version to get some fixes
-libraryDependencies += "org.scalariform" %% "scalariform" % "0.1.5-20140822-69e2e30"
+//libraryDependencies += "org.scalariform" %% "scalariform" % "0.1.8"
```
Comment (schmitch) [inline framework/project/plugins.sbt]: Guess we should remove the out commented stuff, here and the excludeAll, if that works.

**Comment by schmitch** [comment]
besides the out commented stuff I guess that is good to go and also makes the styling more consistent.

**Comment by wsargent** [comment]
@schmitch rebased and ready for review :-)

File: framework/project/Build.scala [hunk 2]
```diff
@@ -47,6 +50,16 @@ object BuildSettings {
    * These settings are used by all projects
    */
   def playCommonSettings: Seq[Setting[_]] = scalariformSettings ++ Seq(
+      ScalariformKeys.preferences := ScalariformKeys.preferences.value
+        .setPreference(SpacesAroundMultiImports, true)
+        .setPreference(SpaceInsideParentheses, false)
+        .setPreference(DanglingCloseParenthesis, Preserve)
+        .setPreference(PreserveSpaceBeforeArguments, true)        
+        .setPreference(DoubleIndentClassDeclaration, true)
+        //.setPreference(DoubleIndentMethodDeclaration, true)
```
Comment (schmitch) [inline framework/project/Build.scala]: should these comments be kept?

File: framework/src/play-netty-server/src/main/scala/play/core/server/netty/NettyModelConversion.scala [hunk 3]
```diff
@@ -167,7 +168,8 @@ private[server] class NettyModelConversion(forwardedHeaderHandler: ForwardedHead
   /** Create a Netty response from the result */
   def convertResult(
     result: Result, requestHeader: RequestHeader, httpVersion: HttpVersion, errorHandler: HttpErrorHandler)(
-      implicit mat: Materializer): Future[HttpResponse] = {
+    implicit
```
Comment (schmitch) [inline framework/src/play-netty-server/src/main/scala/play/core/server/netty/NettyModelConversion.scala]: scalariform isn't clever..

Guess we should split it earlier and hope that it gets it right in that case?

Comment (wsargent) [inline framework/src/play-netty-server/src/main/scala/play/core/server/netty/NettyModelConversion.scala]: Yah, I split it.

File: framework/project/Build.scala (hunk 2 above)
Comment (wsargent) [inline framework/project/Build.scala]: Removed.

**Pull request merged**